build/print.go

```diff
@@ -758,10 +758,26 @@ func (p *printer) seq(brack string, start *Position, list *[]Expr, end *End, mod
 
 		p.newline()
 		p.expr(x, precLow)
-		// Don't print a comma after the last element in modeParen and in modeDef
-		if !(mode == modeDef || mode == modeParen) || i+1 < len(*list) {
-			p.printf(",")
+
+		// Situations where we don't print a final comma
```

Comment:
instead of this comment, make this a well-named method
func needsComma(...) bool

then
if needsComma(...) {
  p.printf(",")
}